### PR TITLE
Add role attributes for accessibility

### PR DESCRIPTION
**Pull request by Ladsgroup**
These are required in ARIA a11y guidelines.
Some info:
https://dequeuniversity.com/rules/axe/3.5/landmark-one-main
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role

### CODE REVIEW

**Comment by jakobw** [comment]
The changes to the headings don't look right to me. Even the article you linked says that the heading structure we had before is actually preferable:

> Instead of using a `<div>` or `<span>` with a heading role and aria-level, consider using a native `<h1>` through `<h6>` element instead to indicate that this text is a heading, and what part of the structure it represents.

-> https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role

That said, adding `role="main"` to the element around the main content, and adding `role="form"` around the element containing the actual query builder seem like the right thing to do.

**Comment by Ladsgroup** [comment]
> The changes to the headings don't look right to me. Even the article you linked says that the heading structure we had before is actually preferable:
> 
> > Instead of using a `<div>` or `<span>` with a heading role and aria-level, consider using a native `<h1>` through `<h6>` element instead to indicate that this text is a heading, and what part of the structure it represents.
> 
> -> https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role
> 
> That said, adding `role="main"` to the element around the main content, and adding `role="form"` around the element containing the actual query builder seem like the right thing to do.

you're right, it's just so complicated (a landmark should not be inside another landmark, every tag should be covered by a landmark, etc.) that sometimes it makes tricky to work it out. With the change I just did, it passes and doesn't put heading on a <div> element. Hope that works now.

File: src/components/QueryBuilder.vue [hunk 1]
```diff
@@ -1,27 +1,29 @@
 <template>
-	<div class="querybuilder">
-		<h1 class="querybuilder__heading">Simple Query Builder</h1>
+	<div class="querybuilder" role="main">
+		<h1 class="querybuilder__heading" role="heading" aria-level="1">Simple Query Builder</h1>
```
Comment (jakobw) [inline src/components/QueryBuilder.vue]: I'm still not convinced. Doesn't https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role say that `<h1>` on its own is equivalent to another element with ` role="heading" aria-level="1"` set? Can't we just leave it as `<h1 class="querybuilder__heading">`?

Comment (Ladsgroup) [reply src/components/QueryBuilder.vue]: oh that part, I read somewhere it's better to keep html5 and ARIA identifiers.
https://dequeuniversity.com/rules/axe/3.5/landmark-one-main
> It is a best practice to use both HTML 5 and ARIA landmarks to ensure all content is contained within a navigational region.

But OTOH, this is not a landmark. I'd remove the heading bit.

**Review by jakobw — APPROVED** [review]
LGTM, thanks!! :sparkles:

**Pull request merged**